languages/go/reference/README.md

```diff
@@ -1,7 +1,11 @@
 # Go
 
+This is a list of concepts of Go. They are categorized into `OOP`, `Functional` and `General`. The `Specialties` category tries to mention concepts that are somewhat special to Go. A concept can belong to multiple categories.
+
 ## Specialties
 
+Concepts that are special to Go or are used differently to a vast majority of other languages.
+
```

Comment:
Would something like `Code Generation` be a fitting example? This is not specific only to go, but is something pretty advanced within the go world.

Reply:
Would be a pretty advanced concept. I like it!